opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -1737,7 +1737,7 @@ namespace Opm {
                 const auto& segments = well.segments;
 
                 // \Note: eventually we need to hanlde the situations that some segments are shut
-                assert(int(segment_set.size()) == segments.size());
+                assert( 1u + segment_set.size() == segments.size());
```

Comment:
Just curious why we need to add 1 here? Is it because other changes somewhere? I am not saying it is wrong, since the tests pass.

Reply:
I now think it is a mistake. We have not got hit by it is because it only called by a restarting case with multi-segment well, with which we do not have a test case yet.

Reply:
Will the original form `int(segment_set.size()) == segments.size()` cause warning with g++-8.0?

Reply:
@blattms